test/meson.build

```diff
@@ -0,0 +1,76 @@
+add_languages('cpp')
+
+gtest_sources = [
+  'gtest/src/gtest-all.cc',
+  'gtest/src/gtest_main.cc',
+]
+
+gtest_includes = [
+  include_directories('gtest'),
+  include_directories('gtest/include'),
+]
+
+libgtest = static_library(
+  'gtest',
+  gtest_sources,
+  include_directories : gtest_includes)
+
+test_i965_headers = [
+  'i965_avce_test_common.h',
+  'i965_config_test.h',
+  'i965_internal_decl.h',
+  'i965_jpeg_test_data.h',
+  'i965_streamable.h',
+  'i965_test_environment.h',
+  'i965_test_fixture.h',
+  'i965_test_image_utils.h',
+  'test.h',
+  'test_utils.h',
+]
+
+test_i965_sources = [
+  'i965_avcd_config_test.cpp',
+  'i965_avce_config_test.cpp',
+  'i965_avce_context_test.cpp',
+  'i965_avce_test_common.cpp',
+  'i965_chipset_test.cpp',
+  'i965_config_test.cpp',
+  'i965_initialize_test.cpp',
+  'i965_jpeg_test_data.cpp',
+  'i965_jpeg_decode_test.cpp',
+  'i965_jpeg_encode_test.cpp',
+  'i965_jpegd_config_test.cpp',
+  'i965_jpege_config_test.cpp',
+  'i965_surface_test.cpp',
+  'i965_test_environment.cpp',
+  'i965_test_fixture.cpp',
+  'i965_test_image_utils.cpp',
+  'object_heap_test.cpp',
+  'test_main.cpp',
+]
+
+gtest_includes = [
+  srcdir,
+  include_directories('gtest/include'),
+]
+
+testdriverdir = join_paths(meson.build_root(), 'src')
+cpp_flags = [
+  '-DTEST_VA_DRIVERS_PATH="' + testdriverdir + '"',
+  '-DGTEST_HAS_PTHREAD=' + thread_dep.found().to_string('1', '0'),
+  '-DGTEST_LANG_CXX11=1',
+  '-DGTEST_USE_OWN_TR1_TUPLE=0',
+  '-DGTEST_HAS_TR1_TUPLE=1',
```

Comment:
> Also, -DPTHREAD?

In src's meson.build there is

```
if thread_dep.found()
  cflags += [ '-DPTHREADS' ]
endif
```

Reply:
Right, but cflags aren't added to any of test's meson.build targets.  If you turn on verbose compile, then you'll see that it is missing during compile of test targets.

Reply:
-DPTHREADS is definitely needed since without it the sizeof(_I965Mutex) will not be the same between i965_drv_video.so and test_i965_drv_video binaries.  This will effect the sizeof(struct object_heap) and sizeof(struct i965_driver_data) also.  Hence, the reason why you see strange behavior with i965_driver_data in tests.  That is, vaInitialize will dlopen the driver, which has its own compiled-in struct definition whereas the test fixture will use its own compiled-in struct definition.

In general, both driver and test executable should always be compiled with exact same compiler defines.  More specifically, any preprocessor definitions on the compiler command line (and config.h) that get used in any of the headers must match for both targets to ensure exact same data structures are defined.

Reply:
As far as I understand, PTHREADS macro is only needed by libi965_drv_video static library and by the driver it self, so, if I understand correctly, there's no need to set it for the tests since they are linking the already compiled library.

Reply:
PTHREADS is used in driver headers and the tests include driver headers.